content/reference/wadm/api.md

```diff
@@ -0,0 +1,218 @@
+
+**Request**: empty, or string containing version
+
+**Response**:
+```json
+{
+    "result": "deleted|error",
+    "undeploy": true,
+    "message": "..."
+}
+```
+
+### Version History
+`wadm.api.{lattice}.model.versions.{name}`
+
+Retrieves a list of all stored versions of a given model, including when that version was created. 
+
+**Request**: empty
+
+**Response**: 
+```json
+[
+    {
+        "version": "...",
+        "created": "RFC3339",
+        "deployed": false
+    }
+]
+```
+
+Each of the items in the response list describes a revision and indicates whether that revision is the deployed one.
+
+
+## Managing Deployments
+Deployments are discrete instances of autonomous control agents that monitor an application model against the real-time, discovered state of a lattice. This agent monitors lattice events, compares lattice state against the desired state of the model, and issues lattice control interface commands in order to compensate for observed gaps.
+
+### Deploy
+`wadm.api.{lattice}.model.deploy.{name}`
+
+Deploys the indicated model. This operation is idempotent, and will simply "succeed" if a request to deploy an already deployed model is received. The previous version, if deployed, does not need to be _undeployed_. Instead, the desired state of the new version is compared against the observed state of the lattice and the autonomous agents will compensate accordingly.
+
+**Request**: `string` containing version to deploy. This version must already exist in model persistence.
+
+**Response**:
+```json
+{
+    "result": "success|error",
+    "message": "..."
+}
+```
+
+It's important to note that the success/fail of this call doesn't indicate completed deployment, only success or failure of the receipt of the deployment request.
+
+### Undeploy
+`wadm.api.{lattice}.model.undeploy.{name}`
+
+Undeploys the currently deployed version of a model. If the model is already undeployed, the operation is idempotent and will return success along with an appropriate message. If the request field `destructive` is true, then the agent will attempt to remove all resources managed by the deployment from the lattice. Otherwise, previously managed resources will remain running.
+
+**Request**: 
+```json
+{
+    "destructive": false
+}
+```
+
+**Response**:
+```json
+{
+    "result": "success|error",
+    "message": "..."
+}
+```
+
+### Deployment Status
+`wadm.api.{lattice}.model.status.{name}`
+
+Describes the deployment status of a given model. Note that since only one version of a model is ever deployed at one time, we don't need to specify a version here. The currently deployed version is included in the response.
+
+The response includes a `status` field which is an aggregate at high levels and an individual status for single traits.
+
+**Request**: `none`
+
+**Response**:
+```json
+{
+    "version": "...",
+    "status": "compensating|ready|failed",
```

Comment:
We probably want a whole `status` object that can contain additional data like messages